From c9eefa180576e7218d6aef063ea52915c0ce18a6 Mon Sep 17 00:00:00 2001
From: David Kupka <dkupka@redhat.com>
Date: Mon, 27 Mar 2017 16:09:09 +0200
Subject: [PATCH] rpcserver.login_x509: Actually return reply from __call__
 method

__call__ didn't return causing internal error in wsgi application. Previously
this bug was hidden by some other error and the code worked even though it
shouldn't.

https://pagure.io/freeipa/issue/6819

Reviewed-By: Pavel Vomacka <pvomacka@redhat.com>
---
 ipaserver/rpcserver.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipaserver/rpcserver.py b/ipaserver/rpcserver.py
index be4e3916b6011dd2b6c90a0267990bf1e370dfb9..77ed7e124c2ca3dcb49d3a68269d6fa9875d4da0 100644
--- a/ipaserver/rpcserver.py
+++ b/ipaserver/rpcserver.py
@@ -842,7 +842,7 @@ class login_x509(KerberosLogin):
                 environ, start_response, 'KRB5CCNAME not set',
                 'Authentication failed')
 
-        super(login_x509, self).__call__(environ, start_response)
+        return super(login_x509, self).__call__(environ, start_response)
 
 
 class login_password(Backend, KerberosSession):
-- 
2.12.1

